phpunit/functional/UserTest.php

```diff
@@ -1012,6 +1012,113 @@ public function testGetFriendlyName($input, $rawname)
         $this->assertSame($rawname, $user->getFriendlyName());
     }
 
+    public function testGetFriendlyNameFields()
+    {
+        global $DB;
+
+        // We'll test both name formats
+        $name_formats = [
+            User::REALNAME_BEFORE  => [
+                'label' => 'REALNAME_BEFORE',
+                'expected' => [
+                    'both' => 'Doe John',
+                    'realname_only' => 'Smith',
+                    'firstname_only' => 'Alice',
+                    'neither' => 'user4'
+                ]
+            ],
+            User::FIRSTNAME_BEFORE => [
+                'label' => 'FIRSTNAME_BEFORE',
+                'expected' => [
+                    'both' => 'John Doe',
+                    'realname_only' => 'Smith',
+                    'firstname_only' => 'Alice',
+                    'neither' => 'user4'
+                ]
+            ]
+        ];
+
+        $this->login();
+
+        // Create test users with different combinations of name/firstname
+        $user_data = [
+            // User with both firstname and realname
+            [
+                'name' => 'user1',
+                'realname' => 'Doe',
+                'firstname' => 'John',
+            ],
+            // User with only realname
+            [
+                'name' => 'user2',
+                'realname' => 'Smith',
+                'firstname' => '',
+            ],
+            // User with only firstname
+            [
+                'name' => 'user3',
+                'realname' => '',
+                'firstname' => 'Alice',
+            ],
+            // User with neither realname nor firstname
+            [
+                'name' => 'user4',
+                'realname' => '',
+                'firstname' => '',
+            ]
+        ];
```

Comment:
It seems each data is tested individually, they do not rely on each others.
Shouldn't this be separated in a data provider instead ?